### PR TITLE
Refactor HTMLLabel

### PR DESCRIPTION
**Pull request by AtishPapa**
# Why?

The current implementation of `HTMLLabel` depends on a solution which dispatches to WebKit when creating the HTML string, causing crashes and layout issues.

# What?

The inner part of the label was remade to use the lexer, also the UIKit based parser was refactored to return an attributed string. Since the basic HTML elements should be the same, currently the custom styling is narrowed to the `style` tag.

# Version Change

Breaking: The styling interface of HTMLLabel changed.

### CODE REVIEW

File: FinniversKit/Sources/Components/HTMLLabel/HTMLLabel.swift [hunk 1]
```diff
@@ -1,26 +1,31 @@
 public class HTMLLabel: Label {
     private var htmlText: String?
-    private var styleMap: HTMLStyler.StyleMap
-    private var additionalStyleMap: HTMLStyler.StyleMap?
+    private let spanMapper: HTMLStringUIKitStyleTranslator.SpanMapper
```
Comment (BjornRuud) [inline FinniversKit/Sources/Components/HTMLLabel/HTMLLabel.swift:3]: In the future this class should probably be modified to accept custom mapping for any element. We have no idea what kinds of HTML the various backends want to send.

Comment (BjornRuud) [reply FinniversKit/Sources/Components/HTMLLabel/HTMLLabel.swift:3]: There are various weights of bold, and in CSS you can style elements however you want regardless of their original intent. We need a default for the common semantic tags though.

**Review by BjornRuud — APPROVED** [review]
LGTM! Nice tests for attributed string output. 👍

File: FinniversKit/Sources/Components/HTMLLabel/HTMLLabel.swift (hunk 1 above)
Comment (AtishPapa) [reply FinniversKit/Sources/Components/HTMLLabel/HTMLLabel.swift:3]: That's fine, but until we don't get that sophisticated HTML syntax from backend, I think it's simpler to only open up `span` at the moment.

**Pull request merged**